common/app/model/ShareLinks.scala

```diff
@@ -13,6 +15,19 @@ case class ShareLink (
   val userMessage: String = platform.userMessage
 }
 
+final case class ShareLinkMeta(
+  visible: Seq[ShareLink],
+  hidden: Option[NonEmptyList[ShareLink]]
+)
+
+object ShareLinkMeta {
+  def noneHidden(shares: ShareLinkMeta): ShareLinkMeta =
+    ShareLinkMeta(shares.visible ++ shares.hidden.fold(Nil: Seq[ShareLink])(_.toList), None)
+
+  def create: ((Seq[ShareLink], Option[NonEmptyList[ShareLink]])) => ShareLinkMeta =
```

Comment:
`apply` is a better name. 👍  A tuple because that is what we have access to later. It's a bit more verbose, but happy to manually pull apart the tuple below